sdk/messaging/azservicebus/internal/errors.go

```diff
@@ -171,10 +171,14 @@ func GetRecoveryKind(err error) recoveryKind {
 		}
 	}
 
-	var me mgmtError
+	var rpcErr rpcError
 
-	if errors.As(err, &me) {
-		code := me.RPCCode()
+	if errors.As(err, &rpcErr) {
+		code := rpcErr.RPCCode()
+
+		if code == 404 {
```

Comment:
At some point would be good to switch to using the constants from the [`net/http`](https://pkg.go.dev/net/http#pkg-constants) package.

Reply:
They do seem to line up with HTTP status codes but they aren't coming back from HTTP in this case. 

I could make some aliases though since it's kind of silly to just have the values hanging out.

Reply:
(going to leave this for now. I need to go actually repro some of these to see the descriptions that come back from the service)